DrivingChassis.cpp

```diff
@@ -69,7 +74,148 @@ DrivingChassis::DrivingChassis(PIDMotor * left, PIDMotor * right,
  * @note pidmotorInstance->overrideCurrentPosition(0); can be used to "zero out" the motor to
  * 		 allow for relative moves. Otherwise the motor is always in ABSOLUTE mode
  */
-void DrivingChassis::driveForward(float mmDistanceFromCurrent, int msDuration) {
+void DrivingChassis::driveForwardFromInterpolation(float mmDistanceFromCurrent, int msDuration) {
+	// We should also have a "drive straight" method that uses the IMU to actually drive straight
+	// Although we'll have a linefollow state so... idk maybe not
+
+	myleft -> overrideCurrentPosition(0);
+	myright -> overrideCurrentPosition(0);
+	myleft -> startInterpolationDegrees(mmDistanceFromCurrent * MM_TO_WHEEL_DEGREES, msDuration, LIN);
+    myright -> startInterpolationDegrees(-mmDistanceFromCurrent * MM_TO_WHEEL_DEGREES, msDuration, LIN);
+}
+
+DrivingStatus DrivingChassis::driveForward(float mmDistanceFromCurrent, int msDuration){
```

Comment:
this should be called once and only once, make a different method to check for completion

Reply:
the functions driveForward, driveBackwards, turnToHeading are meant to run in the updateStateMachine. I'll give an example. Imagine we have a routine which is to drive forward for 300 mm and then turn right.  This routine is called in updateStateMachine. 

in that routine we do something like: 
if(driveForward(300, 5000) == REACHED_SETPOINT){
     routineStatus = turning
}

and then inside of turning, we do the same thing for the turnToHeading. 

Something we can do is make commandDriveForward, and checkDriveForwardStatus. Basically breaking it up like you suggest. In that case we must add a separate state in the state machine for setting the command to drive forward, and another separate state to turn, instead of checking if we've reached the setpoint in the same state as assigning it. Another solution is just setting a boolean that you've set a setpoint. 

This is a lot of text so let me know if this doesn't make sense and I can jump on a call to clarify

Reply:
i do see how that works, i just disagree on that implementation. the splitting into command and check is much more "state machine like" and would be great